docs/source/l1_to_l2b_e2e.rst

```diff
@@ -0,0 +1,220 @@
+    - Bad Pixel Maps: Identifies pixels that are defective or not functioning properly.
+
+
+Setup
+~~~~~~
+
+Before running the test, ensure you have the necessary datasets and calibration files. You will need:
+
+1. **L1 Data** - Raw Level 1 data files.
+2. **Cals** - Calibration files required for processing as outlined above. 
+
+The test script is written to process the L1 data, apply the calibration files, and produce the output in the L2b format.
+
+
+.. code-block:: python
+
+    import argparse
+    import os
+    import pytest
+    import numpy as np
+    import astropy.time as time
+    import astropy.io.fits as fits
+    import corgidrp
+    import corgidrp.data as data
+    import corgidrp.mocks as mocks
+    import corgidrp.walker as walker
+    import corgidrp.caldb as caldb
+    import corgidrp.detector as detector
+
+
+Before processing the L1 files, the TVAC FITS headers must be standardized using the ``fix_headers_for_tvac`` function:
+
+Helper function to standardize TVAC FITS headers for pipeline compatibility:
+
+.. code-block:: python
+
+    
+        def fix_headers_for_tvac(list_of_fits):
+        print("Fixing TVAC headers")
+        for file in list_of_fits:
+                fits_file = fits.open(file)
+                prihdr = fits_file[0].header
+                exthdr = fits_file[1].header
+                prihdr['VISTYPE'] = "TDEMO"
+                prihdr['OBSNUM'] = prihdr['OBSID']
+                exthdr['EMGAIN_C'] = exthdr['CMDGAIN']
+                exthdr['EMGAIN_A'] = -1
+                exthdr['DATALVL'] = exthdr['DATA_LEVEL']
+                exthdr['KGAINPAR'] = exthdr.get('KGAIN', 8.7)
+                prihdr["OBSNAME"] = prihdr['OBSTYPE']
+                prihdr['PHTCNT'] = False
+                exthdr['ISPC'] = False
+                fits_file.writeto(file, overwrite=True)
+
+This function modifies TVAC FITS headers to match the structure expected by the pipeline.
+
+Set up directory paths for input and output data:
+
+.. code-block:: python
+
+    e2edata_path = "/path/to/CGI_TVAC_Data"
+    output_path = os.getcwd()
+
+    l1_datadir = os.path.join(e2edata_path, "TV-36_Coronagraphic_Data", "L1")
+    l2a_datadir = os.path.join(e2edata_path, "TV-36_Coronagraphic_Data", "L2a")
+    l2b_datadir = os.path.join(e2edata_path, "TV-36_Coronagraphic_Data", "L2b")
+    processed_cal_path = os.path.join(e2edata_path, "TV-36_Coronagraphic_Data", "Cals")
+
+    test_outputdir = os.path.join(output_path, "l1_to_l2b_output")
+    l2a_outputdir = os.path.join(test_outputdir, "l2a")
+    l2b_outputdir = os.path.join(test_outputdir, "l2b")
+
+    for d in [test_outputdir, l2a_outputdir, l2b_outputdir]:
+        os.makedirs(d, exist_ok=True)
+
+Define input files and prepare headers:
+
+.. code-block:: python
+
+    l1_data_filelist = [os.path.join(l1_datadir, f"{i}.fits") for i in [90499, 90500]]
+    mock_cal_filelist = [os.path.join(l1_datadir, f"{i}.fits") for i in [90526, 90527]]
```

Comment:
there should be more text explanation about how we're mocking up some fake calibration data here.